test/modules/TestResultCoordinator/Reports/NetworkControllerReportGenerator.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    sealed class NetworkControllerReportGenerator : ITestResultReportGenerator
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(DeploymentTestReportGenerator));
+
+        readonly string trackingId;
+
+        internal NetworkControllerReportGenerator(
+            string trackingId,
+            string source,
+            ITestResultCollection<TestOperationResult> testResults)
+        {
+            this.trackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
+            this.Source = Preconditions.CheckNonWhiteSpace(source, nameof(source));
+            this.TestResults = Preconditions.CheckNotNull(testResults, nameof(testResults));
+            this.ResultType = TestOperationResultType.Network.ToString();
+        }
+
+        internal string Source { get; }
+
+        internal ITestResultCollection<TestOperationResult> TestResults { get; }
+
+        internal string ResultType { get; }
+
+        public async Task<ITestResultReport> CreateReportAsync()
+        {
+            Logger.LogInformation($"Start to generate report by {nameof(NetworkControllerReportGenerator)} for Source [{this.Source}].");
+
+            var results = new List<TestOperationResult>();
+            bool hasResult = await this.TestResults.MoveNextAsync();
+
+            while (hasResult)
```

Comment:
because it always move to next you can use: while (await this.TestResults.MoveNextAsync())